### ISSUE TITLE
Crash at exit when using `headless` with e.g. `doctool` option

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#53289]
### Godot version

4.0 8be97e3b51f5adbee0f9f5d3193da53680a44695

### System information

Ubuntu 21.04, Intel HD 3000

### Issue description

```
godot4 --headless --doctool .
```
cause this crash
```
servers/rendering/renderer_scene_cull.cpp:3767:75: runtime error: member access within null pointer of type 'struct RendererSceneOcclusionCull'

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.dev.custom_build (2d1699ef82a29ef65d810194ee76dddc0d9e0389)
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[1] godot4s(+0x316205ce) [0x55e8b805d5ce] (??:0)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x41040) [0x7ff9db558040] (??:0)
[3] RendererSceneCull::free(RID) (??:0)
[4] RenderingServerDefault::_free(RID) (??:0)
[5] RenderingServerDefault::free(RID) (??:0)
[6] Shader::~Shader() (??:0)
[7] void memdelete<Shader>(Shader*) (??:0)
[8] Ref<Shader>::unref() (??:0)
[9] ColorPicker::finish_shaders() (??:0)
[10] unregister_scene_types() (??:0)
[11] Main::cleanup(bool) (??:0)
[12] godot4s(main+0x415) [0x55e8b805b09e] (??:0)
[13] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xd5) [0x7ff9db53f565] (??:0)
[14] godot4s(_start+0x2e) [0x55e8b805abce] (??:0)
-- END OF BACKTRACE --
================================================================
```

### Steps to reproduce

Above

**Comment by akien-mga** [comment]
Fixed by #53208?

**Comment by qarmin** [comment]
Yes, this PR fixes crash
